frontend/src/pages/NewRun.tsx

```diff
@@ -301,8 +301,13 @@ class NewRun extends Page<{}, NewRunState> {
       try {
         pipeline = await Apis.pipelineServiceApi.getPipeline(pipelineId);
       } catch (err) {
-        await this.showPageError(`Error: failed to retrieve pipeline with ID: ${pipelineId}.`, err);
-        logger.error(`Error: failed to retrieve pipeline with ID: ${pipelineId}`, err);
+        this.setState({ pipelineSelectorOpen: false }, async () => {
+          const errorMessage = await errorToMessage(err);
```

Comment:
Separately, I wonder if `errorToMessage` should be called within `showErrorDialog` too.

Reply:
yes, that probably makes sense. we do it already for page errors